tests/test-helmet.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * Class Test_Helmet
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+use WP_Irving\Templates;
+use function WP_Irving\Templates\create_or_update_title;
+use function WP_Irving\Templates\get_title_from_helmet;
+use function WP_Irving\Templates\inject_head_tags;
+use function WP_Irving\Templates\parse_html;
+use WP_UnitTestCase;
+
+/**
+ * Test helmet functionality.
+ *
+ * @group templates
+ */
+class Test_Helmet extends WP_UnitTestCase {
+
+	/**
+	 * Example output from Yoast's `wpseo_head` action.
+	 *
+	 * @var string
+	 */
+	public $example_markup = '<!-- This site is optimized with the Yoast SEO plugin v14.1 - https://yoast.com/wordpress/plugins/seo/ -->
+<title>Irving Development - Just another WordPress site</title>
+<meta name="description" content="Just another WordPress site" />
+<meta name="robots" content="index, follow" />
+<link rel="canonical" href="https://irving.alley.test/" />
+<meta name="googlebot" content="index, follow, max-snippet:-1, max-image-preview:large, max-video-preview:-1" />
+<meta name="bingbot" content="index, follow, max-snippet:-1, max-image-preview:large, max-video-preview:-1" />
+<script type="application/ld+json" class="yoast-schema-graph">{"@context":"https://schema.org","@graph":[{"@type":"WebSite","@id":"https://irving.alley.test/#website","url":"https://irving.alley.test/","name":"Irving Development","description":"Just another WordPress site","potentialAction":[{"@type":"SearchAction","target":"https://irving.alley.test/?s={search_term_string}","query-input":"required name=search_term_string"}],"inLanguage":"en-US"},{"@type":"CollectionPage","@id":"https://irving.alley.test/#webpage","url":"https://irving.alley.test/","name":"Irving Development - Just another WordPress site","isPartOf":{"@id":"https://irving.alley.test/#website"},"description":"Just another WordPress site","inLanguage":"en-US"}]}</script>
+<!-- / Yoast SEO plugin. -->';
+
+	/**
+	 * Tests the automatic insertion and filters for <Helmet>.
+	 */
+	function test_setup_helmet() {
```

Comment:
These are good tests, but there's a lot going on in this one test method. Let's try and break these up so that generally there is only one assertion (or type of assertion) being made per test method, or reuse the same test method to iterate over different data using a data provider.